### PR TITLE
FT2021_2RC2

### PR DESCRIPTION
**Pull request by blchoy**
First commit of IWXXM 2021-2RC2.  Significant changes include:

1. Implement changes to SIGMET in Amendment 79 [Discussion 267](https://github.com/wmo-im/iwxxm/discussions/267), including the schema/schematron rules and examples.
2. Update examples of SIGMET, VAA and SWA to correct the orientation (counter-clockwise) of the points in a polygon enclosing a phenomenon
3. Change the version of all schemas and examples from RC1 to RC2.

Do not squash and merge this PR.  Only check and approve the changes and leave this branch untouched.

### CODE REVIEW

File: IWXXM/sigmet.xsd [hunk 1]
```diff
@@ -1,497 +1,407 @@
...
-		<attributeGroup ref="gml:AssociationAttributeGroup"></attributeGroup>
-		<attributeGroup ref="gml:OwnershipAttributeGroup"></attributeGroup>
-	</complexType>
-	<simpleType name="SIGMETExpectedIntensityChangeType">
-		<annotation>
-			<documentation>Expected change in intensity for (significant) meteorological phenomena.
+The expected end of occurrence of volcanic ash ("NO VA EXP") is indicated with a missing SIGMET phenomenon with a nil reason of nothingOfOperationalSignificance</xs:documentation>
+						</xs:annotation>
+					</xs:element>
+					<xs:element name="analysisCollection" minOccurs="0" maxOccurs="1">
```
Comment (jkorosi) [inline IWXXM/sigmet.xsd:261]: The maxOccurs should be set to maxOccurs="unbounded" to allow reporting multiple VA clouds. More than one occurrence will be used only by VA SIGMET.

File: IWXXM/WAFSSigWxFC.xsd [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<schema elementFormDefault="qualified" targetNamespace="http://icao.int/iwxxm/2021-2" version="1.0.0RC1" xmlns:iwxxm="http://icao.int/iwxxm/2021-2" xmlns:gml="http://www.opengis.net/gml/3.2" xmlns:aixm="http://www.aixm.aero/schema/5.1.1" xmlns="http://www.w3.org/2001/XMLSchema">
+<schema elementFormDefault="qualified" targetNamespace="http://icao.int/iwxxm/2021-2" version="1.0.0RC2" xmlns:iwxxm="http://icao.int/iwxxm/2021-2" xmlns:gml="http://www.opengis.net/gml/3.2" xmlns:aixm="http://www.aixm.aero/schema/5.1.1" xmlns="http://www.w3.org/2001/XMLSchema">
 	<include schemaLocation="metFeature.xsd"></include>
 	<import namespace="http://www.aixm.aero/schema/5.1.1" schemaLocation="http://www.aixm.aero/schema/5.1.1/AIXM_Features.xsd"></import>
 	<import namespace="http://www.opengis.net/gml/3.2" schemaLocation="http://schemas.opengis.net/gml/3.2.1/gml.xsd"></import>
```
Comment (mgoberfield) [inline IWXXM/WAFSSigWxFC.xsd:5]: I apologize @blchoy for not catching these typos in RC1.
Line 9: weakly-typed, not week-type
Line 180: replace "Oceanic and Atmospheric Administration (NOAA)" with "Weather Service (NWS)"

File: IWXXM/sigmet.xsd [hunk 3]
```diff
@@ -1,497 +1,407 @@
...
+	<xs:element name="VolcanicAshSIGMET" type="iwxxm:VolcanicAshSIGMETType" substitutionGroup="iwxxm:SIGMET">
+		<xs:annotation>
+			<xs:documentation>A SIGMET that reports the presence of volcanic ash conditions hazardous to flight.  This extends the base SIGMET type by including additional information necessary for volcanoes and volcanic ash.</xs:documentation>
+		</xs:annotation>
+	</xs:element>
+	<xs:complexType name="VolcanicAshSIGMETType">
+		<xs:complexContent>
+			<xs:extension base="iwxxm:SIGMETType">
+				<xs:sequence>
+					<xs:element name="supplementaryAnalysisCollection" minOccurs="0" maxOccurs="unbounded">
```
Comment (jkorosi) [inline IWXXM/sigmet.xsd:335]: supplementaryAnalysisCollection should be available for all SIGMET types and used only by VA SIGMET, may be restricted by a Schematron rule. Or only TC SIGMET should be extended by supplementaryAnalysisCollection .

I think a more universal data model is better. It also follows Choy's solution for tropicalCyclonePosition, which is presented in all kind of SIGMETs, but it is used only by TC SIGMET.

**Review by mgoberfield — CHANGES_REQUESTED** [review]

File: IWXXM/iwxxm-collect.xsd [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <schema elementFormDefault="qualified" xmlns="http://www.w3.org/2001/XMLSchema">
-        <import namespace="http://icao.int/iwxxm/2021-2" schemaLocation="http://schemas.wmo.int/iwxxm/2021-2RC1/iwxxm.xsd"/>
+        <import namespace="http://icao.int/iwxxm/3.0" schemaLocation="http://schemas.wmo.int/iwxxm/3.0/iwxxm.xsd"/>
```
[the remark this answers is not in the document]
Comment (blchoy) [reply IWXXM/iwxxm-collect.xsd]: Yes this is an old file.  Fixed in commit https://github.com/wmo-im/iwxxm/pull/272/commits/14b70a04a0e06106377cd927d6dfecf9ca62f08f

File: IWXXM/sigmet.xsd [hunk 5]
```diff
@@ -1,497 +1,407 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<schema elementFormDefault="qualified" targetNamespace="http://icao.int/iwxxm/2021-2" version="3.0.1RC1" xmlns:iwxxm="http://icao.int/iwxxm/2021-2" xmlns:gml="http://www.opengis.net/gml/3.2" xmlns:metce="http://def.wmo.int/metce/2013" xmlns="http://www.w3.org/2001/XMLSchema">
-	<include schemaLocation="common.xsd"></include>
-	<import namespace="http://def.wmo.int/metce/2013" schemaLocation="http://schemas.wmo.int/metce/1.2/metce.xsd"></import>
-	<import namespace="http://www.opengis.net/gml/3.2" schemaLocation="http://schemas.opengis.net/gml/3.2.1/gml.xsd"></import>
-	<annotation>
-		<documentation>SIGMET reporting constructs as defined in ICAO Annex 3 / WMO No. 49-2.
+<!--Generated by Enterprise Architect 12.1.1224 ( Build: 1224 )-->
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" elementFormDefault="qualified" targetNamespace="http://icao.int/iwxxm/2021-2" xmlns:iwxxm="http://icao.int/iwxxm/2021-2" version="4.0.0RC2" xmlns:gml="http://www.opengis.net/gml/3.2" xmlns:metce="http://def.wmo.int/metce/2013">
```
[the remark this answers is not in the document]
Comment (blchoy) [reply IWXXM/sigmet.xsd]: The post-processing script got stuck after I delete some classes in SIGMET.  Fixed in commit https://github.com/wmo-im/iwxxm/pull/272/commits/14b70a04a0e06106377cd927d6dfecf9ca62f08f

File: IWXXM/WAFSSigWxFC.xsd (hunk 2 above)
Comment (blchoy) [reply IWXXM/WAFSSigWxFC.xsd:5]: Line 9: Fixed.
Line 180: I actually get this from WMO Common Code Table C-1 (http://codes.wmo.int/common/centre/177) which is being used in existing BUFR SIGWX files.  As we are not making direct reference to this table I see no reason why we cannot change.  Updated in commit https://github.com/wmo-im/iwxxm/commit/14b70a04a0e06106377cd927d6dfecf9ca62f08f

File: IWXXM/sigmet.xsd (hunk 1 above)
Comment (blchoy) [reply IWXXM/sigmet.xsd]: Both updated in commit https://github.com/wmo-im/iwxxm/commit/14b70a04a0e06106377cd927d6dfecf9ca62f08f

File: IWXXM/sigmet.xsd (hunk 3 above)
Comment (blchoy) [reply IWXXM/sigmet.xsd]: My original intention is to represent the objects before AND with analysisCollection and those after by supplementaryAnalysisCollection.  Your interpretation is obviously different (hence using analysisCollection for VA descriptions before and after AND, and using analysisCollection to represent TC position + CB and supplementaryAnalysisCollection for CB only).  I concur this is a better approach and have implement this in commit https://github.com/wmo-im/iwxxm/commit/14b70a04a0e06106377cd927d6dfecf9ca62f08f

**Review by mgoberfield — APPROVED** [review]
Without commenting on sigmet.xsd, I approve the rest of this PR.

File: IWXXM/WAFSSigWxFC.xsd (hunk 2 above)
Comment (mgoberfield) [reply IWXXM/WAFSSigWxFC.xsd:5]: I stumbled into NOAA vs NWS politics. 😁

File: IWXXM/sigmet.xsd (hunk 3 above)
Comment (blchoy) [reply IWXXM/sigmet.xsd]: After thinking for a while, it seems to me that the location of TC position within the XML document in previous update did not address @jkorosi 's concern that the intensity change is referring to the TC itself and not the CB cloud.  Therefore I have moved the TC position to within <iwxxm:member> and now it should looks exactly like the TAC message; that both TC centre and CB cloud are within the same <iwxxm:member> block described by an @intensityChange.  The revised implementation can be found in commit https://github.com/wmo-im/iwxxm/commit/c29b4a47621b7a942f993e232f44449ea74f88ce

Comment (jkorosi) [reply IWXXM/sigmet.xsd]: I am comfortable with the current proposal.

**Review by jkorosi — APPROVED** [review]

**Comment by blchoy** [comment]
@amilan17 I think we can pack up and upload to schemas.wmo.int right now.

**Comment by mgoberfield** [comment]
HOLD!

Please fix
https://github.com/wmo-im/iwxxm/blob/3b7d03ba782a0afef89060b58623836040002f73/IWXXM/tropicalCycloneAdvisory.xsd#L103

That element needs to be made nillable @blchoy.

**Review by mgoberfield — APPROVED** [review]
Provided the TCA element cumulonimbusCloudLocation is made nillable, I approve.

**Comment by blchoy** [comment]
> Please fix
> 
> https://github.com/wmo-im/iwxxm/blob/3b7d03ba782a0afef89060b58623836040002f73/IWXXM/tropicalCycloneAdvisory.xsd#L103
> 
> That element needs to be made nillable @blchoy. Very sorry that I did not catch this or thought of this yesterday. I realized this omission when I woke up this morning!

Done with commit https://github.com/wmo-im/iwxxm/commit/6136d033723b85cb193a2adbddbd040ee5c8dd8f

Some people say their minds keep figuring out solutions when they are sleeping and will most likely get one when they wake up.  Now I know this is true.  ;)

**Comment by blchoy** [comment]
A new commit https://github.com/wmo-im/iwxxm/pull/272/commits/3bff2a4b1e1ec7721ea083e6ecfc14a382fd0809 was made to handle the remaining documentation issues in #250 and #270.

**Comment by blchoy** [comment]
Another commit https://github.com/wmo-im/iwxxm/pull/272/commits/d06fa3f815e71ae619bf6f07d5acaa91a8224527 was made for removing the schematron rule mentioned in #250.

**Comment by blchoy** [comment]
Seems to me that most issues relevant to IWXXM 2021-2RC2 have been fixed.  The last one could be #271, and it will be done with a new branch FT2021-RC2-refactor.  This PR is therefore closed.

**Pull request closed without merging**